### PR TITLE
legion: drop cray-pmi dependency

### PR DESCRIPTION
**Pull request by rbberger**
There are other ways to enforce cray-pmi being loaded in environments that use cray-mpich. This avoids breaking environments where this was already the case and avoids forcing them to manually declare an external.

@elliottslaughter @bonachea

See discussion in #39614

### CODE REVIEW

**Review by elliottslaughter — APPROVED** [review]
Looks good to me.

**Comment by bonachea** [comment]
This entire mess arose because (for legacy reasons) the defaulting logic in GASNet configure required the `cray-pmi` module to be loaded in order for it to automatically detect the location of the Cray PMI library. This module was always loaded by default on Cray XC systems (so it was a reasonable assumption there), but some HPE Cray EX systems have removed it from the default module set, breaking that assumption.

I've just now upgraded GASNet's configure logic in [52de04d](https://bitbucket.org/berkeleylab/gasnet/commits/52de04ddbcaa7f56af1a39a620f6e9ecff97de84) to be smarter and automatically probe `/opt/cray/pe/pmi/default/`, because standard installs of recent versions of Cray PMI are always available there. With this change in place GASNet's configure should correctly automatically detect the Cray PMI prefix, _regardless_ of whether the `cray-pmi` module is loaded or not. This improvement is currently on GASNet's `develop` branch, and will appear in the forthcoming release in the next few weeks.

TL;DR: cray-pmi remains a real dependency, but it's a standard part of the Cray programming environment (which Spack is unable to install) and with the forthcoming version of GASNet the module becomes irrelevant, so this PR to remove the forced external dependency is the "right change".

CC: @PHHargrove

**Review by bonachea — APPROVED** [review]

**Review by alalazo — APPROVED** [review]
